presto-main/src/test/java/com/facebook/presto/util/TestGraphvizPrinter.java

```diff
@@ -0,0 +1,128 @@
+ * limitations under the License.
+ */
+package com.facebook.presto.util;
+
+import com.facebook.presto.common.predicate.TupleDomain;
+import com.facebook.presto.cost.StatsAndCosts;
+import com.facebook.presto.spi.ConnectorId;
+import com.facebook.presto.spi.ConnectorTableHandle;
+import com.facebook.presto.spi.TableHandle;
+import com.facebook.presto.spi.plan.PlanNode;
+import com.facebook.presto.spi.plan.PlanNodeId;
+import com.facebook.presto.spi.plan.TableScanNode;
+import com.facebook.presto.sql.planner.Partitioning;
+import com.facebook.presto.sql.planner.PartitioningScheme;
+import com.facebook.presto.sql.planner.PlanFragment;
+import com.facebook.presto.sql.planner.SubPlan;
+import com.facebook.presto.sql.planner.plan.PlanFragmentId;
+import com.facebook.presto.testing.TestingMetadata.TestingTableHandle;
+import com.facebook.presto.testing.TestingTransactionHandle;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static com.facebook.presto.metadata.FunctionAndTypeManager.createTestFunctionAndTypeManager;
+import static com.facebook.presto.operator.StageExecutionDescriptor.ungroupedExecution;
+import static com.facebook.presto.sql.planner.SystemPartitioningHandle.SINGLE_DISTRIBUTION;
+import static com.facebook.presto.sql.planner.SystemPartitioningHandle.SOURCE_DISTRIBUTION;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.util.GraphvizPrinter.printDistributed;
+import static com.facebook.presto.util.GraphvizPrinter.printLogical;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static org.testng.Assert.assertEquals;
+
+public class TestGraphvizPrinter
+{
+    private static final ConnectorId TEST_CONNECTOR_ID = new ConnectorId("connector_id");
+    private static final PlanNodeId TEST_TABLE_SCAN_NODE_ID = new PlanNodeId("plan_id");
+    private static final ConnectorTableHandle TEST_CONNECTOR_TABLE_HANDLE = new TestingTableHandle();
+    private static final PlanNode TEST_TABLE_SCAN_NODE = new TableScanNode(
+            TEST_TABLE_SCAN_NODE_ID,
+            new TableHandle(TEST_CONNECTOR_ID, TEST_CONNECTOR_TABLE_HANDLE, TestingTransactionHandle.create(), Optional.empty()),
+            ImmutableList.of(),
+            ImmutableMap.of(),
+            TupleDomain.all(),
+            TupleDomain.all());
+    private static final String TEST_TABLE_SCAN_NODE_INNER_OUTPUT = format(
+            "label=\"{TableScan[TableHandle \\{connectorId='%s', connectorHandle='%s', layout='Optional.empty'\\}]}\", style=\"rounded, filled\", shape=record, fillcolor=deepskyblue",
+            TEST_CONNECTOR_ID,
+            TEST_CONNECTOR_TABLE_HANDLE);
+
+    @Test
+    public void testPrintLogical()
+    {
+        String actualStr = printLogical(
```

Comment:
as per our style guidelines, do not use abbreviations. "actual" and "expected" is fine here

Reply:
done